flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/HsSubpartitionConsumerMemoryDataManager.java

```diff
@@ -44,7 +44,6 @@ public class HsSubpartitionConsumerMemoryDataManager implements HsDataView {
     @GuardedBy("consumerLock")
     private final Deque<HsBufferContext> unConsumedBuffers = new LinkedList<>();
 
-    @GuardedBy("consumerLock")
     private final AtomicInteger backlog = new AtomicInteger(0);
```

Comment:
Indeed, this `guardedBy` is required. We have already mark ` @SuppressWarnings("FieldAccessNotGuarded")` for `getBacklog` as it is unsynchronized by design. All other places should be protected by lock, so I still think this field is no need atomic.

Reply:
Fixed.